### PR TITLE
Convert setters from v1 to v2

### PR DESCRIPTION
**Pull request by jlewi**
* Get rid of enable-services.yaml as we are now using CNRM and this is
  in the base package.
* Related to #89

* Remove cluster and iam policy patch; these are now in the base manifests.
  Added in kubeflow/manifests#1398

* kubeflow/hack/create_kptfile.py contains most of the commands
  used to generate the setters and substitutions.

* Add a presubmit to verify the deploy pipeline is working.

### CODE REVIEW

**Comment by jlewi** [comment]
/assign @Ark-kun 

@Ark-kun Could you review it since its very late in shanghai?

File: tests/pipeline-runs/deploy-blueprint-pipeline-run.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      params:
+        - name: revision
+          value: master
+        - name: url
+          value: https://github.com/kubeflow/gcp-blueprints.git
+  - name: testing-repo
+    resourceSpec:
+      type: git
+
+      # TODO(https://github.com/kubeflow/testing/pull/740): Change back
```
Comment (Bobgy) [inline tests/pipeline-runs/deploy-blueprint-pipeline-run.yaml:30]: nit kubeflow/testing#740 already merged

File: kubeflow/hack/create_kptfile.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
+"""This is a script created for the updating of the GCP Kpt packages.
+
+The purpose of this is to upgrade to using setters and substitutions
```
Comment (Bobgy) [inline kubeflow/hack/create_kptfile.py:3]: nit: shall we add a comment to mention it should be in sync with kubeflow/manifest/hack/create_kptfile.py

Comment (jlewi) [reply kubeflow/hack/create_kptfile.py:3]: I think we can delete this in a subsequent PR; I just want to check it in so we have a record in source control but its only intended for one time use.

Comment (Bobgy) [reply kubeflow/hack/create_kptfile.py:3]: SGTM

File: kubeflow/instance/kustomize/kubeflow-apps/service-accounts.yaml [hunk 3]
```diff
@@ -3,4 +3,4 @@ kind: ServiceAccount
 metadata:
   name: kf-admin
   annotations:
-    iam.gke.io/gcp-service-account: kf-kcc-0415-admin@PROJECT.iam.gserviceaccount.com # {"type":"string","x-kustomize":{"setBy":"kpt","partialSetters":[{"name":"name","value":"kf-kcc-0415-001"},{"name":"gcloud.core.project","value":"PROJECT"}]}}
+    iam.gke.io/gcp-service-account: kf-kcc-0415-admin@PROJECT.iam.gserviceaccount.com
```
Comment (Bobgy) [inline kubeflow/instance/kustomize/kubeflow-apps/service-accounts.yaml:6]: @jlewi just noticed this is probably a mistake, there is no setter here

**Pull request merged**